Mathlib/CategoryTheory/Core.lean

```diff
@@ -79,6 +83,94 @@ def forgetFunctorToCore : (G ⥤ Core C) ⥤ G ⥤ C :=
 
 end Core
 
+section
+
+namespace Functor
+
+variable {D : Type u₂} [Category.{v₂} D]
+
+/-- A functor `C ⥤ D` induces a functor `Core C ⥤ Core D`. -/
+@[simps!]
+def core (F : C ⥤ D) : Core C ⥤ Core D := Core.functorToCore (Core.inclusion _ ⋙ F)
+
+variable (C) in
+/-- The core of the identity functor is the identity functor on the cores. -/
+@[simps! hom_app inv_app]
+def coreId : (𝟭 C).core ≅ 𝟭 (Core C) := Iso.refl _
+
+/-- The core of the composition of F and G is the composition of the cores. -/
+@[simps! hom_app inv_app]
+def coreComp {E : Type u₃} [Category.{v₃} E] (F : C ⥤ D) (G : D ⥤ E) :
+  (F ⋙ G).core ≅ F.core ⋙ G.core := Iso.refl _
+
+end Functor
+
+namespace Iso
+
+variable {D : Type u₂} [Category.{v₂} D]
+
+/-- A natural isomorphism of functors induces a natural isomorphism between their cores. -/
+@[simps! hom_app inv_app]
+def core {F G : C ⥤ D} (α : F ≅ G) : F.core ≅ G.core :=
+  NatIso.ofComponents
+    (fun x ↦ Groupoid.isoEquivHom _ _|>.symm <| α.app x)
+    (fun {_ _} f ↦ by aesop_cat)
+
+@[simp]
+lemma coreComp {F G H : C ⥤ D} (α : F ≅ G) (β : G ≅ H) : (α ≪≫ β).core = α.core ≪≫ β.core := rfl
+
+@[simp]
+lemma coreId {F : C ⥤ D} : (Iso.refl F).core = Iso.refl F.core := rfl
+
+-- Arguments order and simp normal forms here are chosen to mimic simp lemmas of pseudofunctors.
+
+@[simp]
```

Comment:
Do we really want all of these to be simp lemmas?

Reply:
They are simp lemmas for pseudofunctors (or did I misunderstand what line `85` of `CategoryTheory/Bicategory/Functor.Pseudofuncto` means?).
My rationale was that even though we can’t really make construction like this actual pseudofunctors due to size shenanigans, the API should be as close as possible, and so have the same simp lemmas.

Reply:
`Iso.core` already has simp lemmas `hom_app/inv_app` attached to it, then I do not think that we should have simp lemmas at the "higher level". These are good lemmas, but I do not think it should be simp lemmas.
(Also, the notion of pseudofunctors is not very much developed in mathlib, and I believe we are not completely sure about what should be the normal forms there.)

Reply:
Got it. I’ll note that this is the current policy for "higher" lemmas and remove the simp tags here.